src/miri.md

```diff
@@ -0,0 +1,47 @@
+# Miri
+
+Miri (**MIR** **I**nterpreter) is a virtual machine for executing MIR without
```

Comment:
Those two are not exclusive, as many stack machines *are also* RAM machines (and if you're a RAM machine, adding a stack is easy).

*However*, miri isn't either of those, as each virtual allocation forms its own "address space", and the stack frames have fixed shapes (determined by the MIR of the fn body/const initializer being evaluated), instead of having more general data stack manipulation primitives.

Reply:
So the stack is implemented more as a linked list of `Frame` in some sense (where each `Frame` is created based on what you are miri-ing)?

Reply:
Well it's a stack of `Frame`s within one evaluation - I don't believe nested `Frame`s are ever created except for `const fn` calls - if you need the value of a different `const`, it would be evaluated separately, with an entirely different stack.

Reply:
Got it :)

It would be good to add this info to the chapter.